pilot/pkg/xds/ecds.go

```diff
@@ -71,5 +117,127 @@ func (e *EcdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 			Resource: util.MessageToAny(c),
 		})
 	}
+
 	return resources, model.DefaultXdsLogDetails, nil
 }
+
+func (e *EcdsGenerator) GeneratePullSecrets(proxy *model.Proxy, updatedSecrets map[model.ConfigKey]struct{}, secretResources []SecretResource,
```

Comment:
The difference from sds is that there is no authorize here, isn's there a risk of secret leak

Reply:
WasmPlugin is a bit different from gateway secret, that root namespace WasmPlugin config could apply to all namespaces, so specify that the secret has to be in the same namespace as the WasmPlugin.